### PR TITLE
Identify OpenAPI

### PR DESCRIPTION
**Pull request by klokane**

### CODE REVIEW

**Comment by kylef** [comment]
:+1:

@klokane Could you add a release commit by bumping the version in `package.json` to `0.4.0`.

I think it may be worth adding a test for other `openapi` version numbers too. Such as `"openapi": "3.1.0"`, the test only test `3.0.0` while other numbers with same major are supported. The code does support it but I think an explicit test would go a long way for it.

File: test/identify-test.js [hunk 1]
```diff
@@ -548,3 +548,171 @@ describe('Refract API Description namespace', function() {
...
+
+describe('OpenAPI', function() {
+  describe('OpenAPI file with arbitrary valid JSON content', function() {
+    const source = dedent`
+      {
+        "openapi": "3.0.0",
+        "info" : {
+           "title" : "api",
+           "version" : "1.0.0"
+	},
```
Comment (pksunkara) [inline test/identify-test.js]:
```suggestion
	      },
```

File: test/identify-test.js [hunk 2]
```diff
@@ -548,3 +548,171 @@ describe('Refract API Description namespace', function() {
...
+  });
+
+  describe('OpenAPI file with arbitrary valid JSON content and multiple spaces between "openapi" key, colon and value', function() {
+    const source = dedent`
+      {
+        "openapi"   :    "3.0.0",
+        "info" : {
+           "title" : "api",
+           "version" : "1.0.0"
+	},
```
Comment (pksunkara) [inline test/identify-test.js]:
```suggestion
	      },
```

File: test/identify-test.js [hunk 3]
```diff
@@ -548,3 +548,171 @@ describe('Refract API Description namespace', function() {
...
+    });
+  });
+
+  describe('OpenAPI file with arbitrary valid JSON content, but openapi key is later in document', function() {
+    const source = dedent`
+      {
+        "info" : {
+           "title" : "api",
+           "version" : "1.0.0"
+	},
```
Comment (pksunkara) [inline test/identify-test.js]:
```suggestion
	      },
```

**Comment by klokane** [comment]
Fixed

File: src/deckardcain.js [hunk 4]
```diff
@@ -7,8 +7,8 @@ export const LEGACY_BLUEPRINT_TITLE = /[-]{3}(?=( [^\n\r]+ | )[-]{3}([\n\r]{1,2}
 export const SWAGGER_JSON = /^[\uFEFF]?{[\s\S]*["']swagger["']\s*:\s*["']\d\.\d["'],?/i;
 export const SWAGGER_YAML = /(?:^|\n)\s*swagger: ["']\d\.\d["']\n/i;
 
-export const OPENAPI_JSON = /^[\uFEFF]?{[\s\S]*["']openapi["']\s*:\s*["']\d\.\d+.\d+["'],?/i;
-export const OPENAPI_YAML = /(?:^|\n)\s*openapi: ["']\d\.\d+.\d+["']\n/i;
+export const OPENAPI_JSON_3_0_0 = /^[\uFEFF]?{[\s\S]*["']openapi["']\s*:\s*["']3.0.0["'],?/i;
```
[the remark this answers is not in the document]
Comment (klokane) [reply src/deckardcain.js]: Sorry. My inattention Wil, fix.

**Comment by pksunkara** [comment]
Please release

**Pull request merged**